jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java

```diff
@@ -382,26 +384,26 @@ private void readAnnotations( Class<?> testClass, Method method ) {
 
     public void addTags( Annotation... annotations ) {
         for( Annotation annotation : annotations ) {
-            List<Tag> tags = toTags( annotation );
-            this.reportModel.addTags( tags );
-            this.scenarioModel.addTags( tags );
+            addTags( toTags( annotation ) );
         }
     }
 
+    private void addTags( List<Tag> tags ) {
+        this.reportModel.addTags( tags );
+        this.scenarioModel.addTags( tags );
```

Comment:
Because the normal path of adding tags via annotation does the same. Which one shouldn't it be added to?